### DISCUSSION TITLE
issue: Watched arrays of objects does not update their reference after object field update [useFieldArray]

### DISCUSSION THREAD

**Discussion by DanielMarkiel** [react-hook-form/react-hook-form#10929] [Q&A] [answered]
### Version Number

7.46.1

### Codesandbox/Expo snack

https://codesandbox.io/s/ecstatic-merkle-cdqjgg?file=/src/App.tsx

### Steps to reproduce

1. Go to https://codesandbox.io/s/ecstatic-merkle-cdqjgg?file=/src/App.tsx
2. Change text in the 'lastName' input
3. Memoized list of last names below input does not update (reference of the `testArray` const returned by 'watch' function stays the same)

### Expected behaviour

Memoized values built upon watched arrays should update when a field of an object in the array changes.

The last version of react-hook-form where this works correctly is 7.13.0. If I downgrade the package to this version the list updates correctly.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@DanielMarkiel 

This is the expected behavior for `watch()` method on field array of react-hook-form latest version and it has been documented in here: 

https://react-hook-form.com/docs/useform/watch

<img width="1050" alt="Screenshot 2023-09-14 at 15 43 31" src="https://github.com/react-hook-form/react-hook-form/assets/18494222/7d9a5101-9d60-424a-ac9f-c14abed7091e">

Field array data will be the same reference array inside form state to avoid re-rendering therefore using `useMemo` or `useEffect` with change deps will not work.

Just remove the use of `useMemo` to achieve your expected behavior.

Working demo: https://codesandbox.io/s/epic-swanson-gtqhsx?file=/src/App.tsx
